### DISCUSSION TITLE
Issues upgrading from 1.3.16

### DISCUSSION THREAD

**Discussion by corunner** [jamessimone/apex-rollup#298] [Q&A]
I just took over an org and would like to upgrade from v1.3.16. I tried to upgrade to 1.4.13. I received an email with 13 different Apex compile failures. It looks like things are missing, so I'm thinking I need to upgrade to a less-recent version before getting up to the most recent one. Could that be the case, and if so, where can I find the install packages for older version?

Here's an example of one of the errors, they all look very similar: 
3.  Apex compile failure
Apex class RollupRelationshipFieldFinderTests: line 29, column 43: Method does not exist or incorrect signature: void retrieveParent(Id) from the type RollupRelationshipFieldFinder.Traversal

Thanks so much.

**Comment by jamessimone** [comment]
@corunner you should be able to install [v1.4.10](https://login.salesforce.com/packaging/installPackage.apexp?p0=04t6g000008Sjv4AAC) first, prior to upgrading. This was the release _prior_ to changing the method signature within `RollupRelationshipFieldFinder.Traversal.retrieveParent` to `...retrieveParents`. You may need to go into the advanced options while installing this specific package, and choose the "Deprecate" option when installing. That should allow you to upgrade without any issues to the more recent versions.

My apologies - I didn't realize there would be a breaking change here, I would have updated the minor version number to `1.5.0` if I had. Let me know if you need any further help!

**Reply by corunner** [reply]
Great, thanks for the fast response. I'm about to be on the move but will try this later today, will report back.

**Reply by corunner** [reply]
I tried v14.10 and unfortunately it still failed with similar errors. I'm attaching a screen shot of the message. Maybe I need to go further back in versions? 
![Screen Shot 2022-04-20 at 9 29 51 AM](https://user-images.githubusercontent.com/104042052/164267722-fe503390-5c8d-4ef0-b436-2b0692653be6.png)

**Reply by jamessimone** [reply]
@corunner interesting, it looks like that property was added in [v1.3.23](https://test.salesforce.com/packaging/installPackage.apexp?p0=04t6g000008SitNAAS) - are you able to install that version?

**Comment by corunner** [comment]
Still having the same issue. I didn't realize we had such an old version (I'm new on the project). Attaching the error screen shot. 
![Screen Shot 2022-04-20 at 11 02 19 AM](https://user-images.githubusercontent.com/104042052/164284521-c6bfd0c4-a8f8-4728-bd53-2fbaf211837b.png)

**Reply by jamessimone** [reply]
@corunner 🤦‍♂️  should have had you try installing the version immediately _before_ those changes were made: https://test.salesforce.com/packaging/installPackage.apexp?p0=04t6g000008SijvAAC

As an aside, if you are continuing to have upgrade issues, it might be easiest to make a copy of your Rollup CMDT records and uninstall/reinstall. I only suggest that as a last resort - I know there _is_ a safe upgrade path, it may just take some wandering around to find. For reference, the link above is to `v1.3.22`, and you can find _all_ the install links by using the [Tags search](https://github.com/jamessimone/apex-rollup/tags) and scrolling to the appropriate version; the README that displays for each version will have the appropriate install link. Hope that helps!

**Comment by corunner** [comment]
Very helpful to know where to find the older install versions, thank you. I was able to update to 1.3.22 but still having the same issue with the next release, 1.3.23. Unless you have another idea for me to try, I'm going to try the uninstall option. We have a lot of flows referencing the CMDTs. I just spun up a new dev sandbox to try this. Thanks again for your help.

**Reply by jamessimone** [reply]
I am of the mind that installing using the `DeprecateOnly` option under advanced options would allow you to upgrade without these issues, and I heartily apologize if that's not the case. Either way, you have my respect for your resourcefulness and understanding!

**Reply by jamessimone** [reply]
Also, if you're familiar with SFDX at all, the CLI-based command options for installing packages with the `DeprecateOnly` flag is a bit more forgiving than having to wend through the Package menus: `sfdx force:package:install -p 04t6g000008SijvAAC -t DeprecateOnly -w 10`, for instance
